front/lib/auth.ts

```diff
@@ -293,6 +294,78 @@ export class Authenticator {
     });
   }
 
+  /**
+   * Get an Authenticator from an auth0 token a given workspace.
+   * This is to be used from the extension, calling our public API.
+   * @param key The Auth0 token
+   * @param wId string the target workspaceId
+   * @returns an Authenticator for wId and the key's own workspaceId
+   */
+  static async fromAuth0Token(
+    token: string,
+    wId: string | undefined
+  ): Promise<Authenticator> {
+    const user = await getUserFromAuth0Token(token);
+
+    let workspace: Workspace | null = null;
+    let role = "none" as RoleType;
+    let groups: GroupResource[] = [];
+    let subscription: SubscriptionType | null = null;
+    let flags: WhitelistableFeature[] = [];
+
+    if (user && typeof wId === "string") {
+      workspace = await Workspace.findOne({
+        where: {
+          sId: wId,
+        },
+      });
+    } else if (user) {
+      const memberships = user
+        ? await MembershipResource.getActiveMemberships({
+            users: [user],
+          })
+        : null;
+      // TODO(EXT): This is a temporary solution to get the workspace from the user's memberships.
```

Comment:
A user can be in many workpsaces. I think we want `wId` to be compulsory here.
As commented below for sessions we have getUserFromSession for workspace agnostic endpoints. we could have a `getUserFromAuth0Token` for similar /v1/ endpoints. This allows us to enforce the fact that an authenticator is always scoped to a workspace.

Reply:
All this have been removed, we're now indeed returning usertype with workspaces and the extension will let the user choose their workspaces at login if multiple memberships are active.